scipy/stats/_continuous_distns.py

```diff
@@ -5959,6 +5959,48 @@ def _entropy(self):
 semicircular = semicircular_gen(a=-1.0, b=1.0, name="semicircular")
 
 
+class sine_gen(rv_continuous):
+    r"""A sine continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The sine distribution is useful for polar angles.
+    The probability density function for `sine` is:
+
+    .. math::
+
+        f(x) = \frac{1}{2} \sin(x)
+
+    for :math:`0 \le x \le \pi`.
+
+    %(after_notes)s
+
+    %(example)s
+
+    """
+    def _pdf(self, x):
+        # sin.pdf(x) = 1/2 * sin(x)
+        return 1.0/2*(np.sin(x))
+
+    def _rvs(self):
+        z = self._random_state.uniform(size=self._size)
```

Comment:
Can you add _ppf method as well? (That could actually be used to generate rvs using the default method.)

Reply:
Added _ppf, and removed _rvs